core/src/main/kotlin/behavior/interaction/ComponentInteractionBehavior.kt

```diff
@@ -0,0 +1,162 @@
+package dev.kord.core.behavior.interaction
+
+import dev.kord.common.annotation.KordPreview
+import dev.kord.common.entity.InteractionResponseType
+import dev.kord.common.entity.MessageFlag
+import dev.kord.common.entity.MessageFlags
+import dev.kord.common.entity.Snowflake
+import dev.kord.common.entity.optional.Optional
+import dev.kord.core.Kord
+import dev.kord.core.supplier.EntitySupplier
+import dev.kord.core.supplier.EntitySupplyStrategy
+import dev.kord.rest.builder.interaction.UpdateMessageInteractionResponseCreateBuilder
+import dev.kord.rest.json.request.InteractionApplicationCommandCallbackData
+import dev.kord.rest.json.request.InteractionResponseCreateRequest
+import kotlin.contracts.ExperimentalContracts
+import kotlin.contracts.InvocationKind
+import kotlin.contracts.contract
+
+@KordPreview
+interface ComponentInteractionBehavior : InteractionBehavior {
+
+    /**
+     * Acknowledges a component interaction publicly with the intent of updating it later.
+     *
+     * There is no requirement to actually update the message later, calling this is
+     * sufficient to handle the interaction and stops any 'loading' animations in the client.
+     *
+     * There is no noticeable difference between this and [acknowledgeEphemeralDeferredMessageUpdate]
+     * when it comes to acknowledging the interaction, both functions can be called
+     * on public and ephemeral messages. The only difference is in the **followUp** calls,
+     * which will become public or ephemeral respectively.
+     */
+    @OptIn(ExperimentalContracts::class)
+    suspend fun acknowledgePublicDeferredMessageUpdate(): PublicInteractionResponseBehavior {
+        val request = InteractionResponseCreateRequest(
+            type = InteractionResponseType.DeferredUpdateMessage
```

Comment:
I see we have changed how we handle requests.
In slash commands I've made the toRequest function determine what's the type response based on input

Reply:
I'm aware, I didn't really see the use of a builder if we can build the request directly anyhow.

Reply:
if that's the case then it'd be better to change the application command to match components